### DISCUSSION TITLE
Running on AWS

### DISCUSSION THREAD

**Discussion by rclark** [wolveix/satisfactory-server#172] [Show and tell]
First off, thank you for this great project! 

I just wanted to share how I use this Docker image to run a dedicated server on AWS for my friends and I to play on. [More details and a CloudFormation template file are available here](https://gist.github.com/rclark/057059bfbd869743d1742a95b456bcff), if you'd like to replicate the setup for yourself.

The highlights:

- Runs on ECS Fargate, so it doesn't rely on any always-on EC2s to host the Docker container. It also uses Spot to keep the costs as low as possible.
- Because containers will stop and be restarted when the application inevitably crashes, the gamefiles and save files are persisted on EFS, where they can be shared from one container to the next.
- Save files are also backed up to S3 on a daily basis, where they are a little easier to access in case you want to grab them.
- When the containers stop and restart, they get new public IP addresses. Instead of using a Load Balancer, which would cost more but provide a consistent DNS name for your dedicated server, a Lambda function manages a DNS `A` record, keeping it up-to-date with the currently-running container's public IP address. 

All in all, it costs about $50-60 per month if you run it 24/7, much less if you turn the container off when no one is playing.

**Comment by wolveix** [comment]
Love this! Thank you so much for sharing @rclark :D
